gax-grpc/src/main/java/com/google/api/gax/grpc/OperationPollingCallable.java

```diff
@@ -99,16 +96,14 @@ public TimedAttemptSettings createNextAttempt(
 
     @Override
     public boolean shouldRetry(Throwable prevThrowable, Operation prevResponse) {
-      return prevThrowable == null && prevResponse != null && !prevResponse.getDone();
-    }
-
-    @Override
-    public boolean shouldCancel(Throwable prevThrowable, Operation prevResponse) {
-      if (prevThrowable == null && prevResponse != null && prevResponse.getError() != null) {
-        Status status = Status.fromCodeValue(prevResponse.getError().getCode());
-        return status.getCode().equals(Status.Code.CANCELLED);
+      if (prevThrowable != null || prevResponse == null) {
+        return false;
       }
-      return false;
+      //      Status status = Status.fromCodeValue(prevResponse.getError().getCode());
+      //      if (Status.Code.CANCELLED.equals(status.getCode())) {
+      //        throw new CancellationException();
+      //      }
```

Comment:
Yes, it is, deleting it. This also implies that in case if client tries to poll an operation which is cancelled on the server side it will get an ApiException (instead of CancellationException) from OperationFuture.get(). This was done on purpose, since ApiException provides information about what actually happened (server canceled operation), while CancellationException does not allow the client to distinguish between him trying to poll a dead operation and him cancelling its own client-side future only. 

As result, OperationFuture cancellation now mean only client-side cancellation, the server side cancellation will result in OperationFuture failure, with ApiException.getCode() == Code.CANCELLED.